src/Microsoft.DotNet.ProjectJsonMigration/MigrationBackupPlan.cs

```diff
@@ -83,8 +85,22 @@ public void PerformBackup()
             {
                 file.MoveTo(
                     Path.Combine(
-                        ProjectBackupDirectory.FullName, file.Name));
+                        ProjectBackupDirectory.FullName,
+                        file.Name));
+            }
+        }
+
+        private static string GetUniqueDirectoryPath(string directoryPath)
+        {
+            var candidatePath = directoryPath;
+
+            var suffix = 1;
+            while (Directory.Exists(directoryPath))
+            {
+                candidatePath = $"{directoryPath}_{suffix++}";
```

Comment:
I was thinking about @piotrpMSFT's concern last night a bit. I believe his concern was around the following scenario:

1. Customer executes 'dotnet migrate' in a folder and 'backup' is placed in parent folder.
2. Customer executes 'dotnet migrate' in a sibling folder to the first one and 'backup_1' is placed in the parent folder.

If I'm remembering correctly, Peter was concerned that the user would have no way to map which 'backup' went with which migration. I believe this isn't a significant problem now that the backup folder is not created in a parent folder. As you said, the latest migration is the one with the highest number.